runtime/src/main/scala/akka/grpc/ServiceObject.scala

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.grpc
+
+import akka.annotation.ApiMayChange
+
+import com.google.protobuf.Descriptors.FileDescriptor;
+
+@ApiMayChange
+trait ServiceObject {
```

Comment:
Not a great name. Something `WithDescriptor` `SelfDescribing` `ServiceMetadata`?